core/src/test/scala/unit/kafka/raft/KafkaNetworkChannelTest.scala

```diff
@@ -159,6 +167,29 @@ class KafkaNetworkChannelTest {
     }
   }
 
+  @ParameterizedTest
+  @MethodSource(Array("fetchVersions"))
+  def testFetchRequestDowngrade(version: Short): Unit = {
+    val destinationId = 2
+    val destinationNode = new Node(destinationId, "127.0.0.1", 9092)
+    channel.updateEndpoint(destinationId, new InetAddressSpec(
+      new InetSocketAddress(destinationNode.host, destinationNode.port)))
+    val mockSendThread = mock(classOf[RaftSendThread])
+    channel.setRequestThread(mockSendThread)
```

Comment:
As said earlier, I would rather prefer to avoid this.

Reply:
Done.